### PR TITLE
Update SonarScanner names and versions

### PR DESCRIPTION
**Pull request by joe-tingsanchali-sonarsource**
Many of the examples in this project refer to "SonarQube" when really any of these examples are applicable for SonarCloud as well. In any case, I tried to label by removing "SonarQube" where possible. I kept SonarQube links.

- Rename the "SonarQube Scanner" to "SonarScanner"
- Update links to match https://docs.sonarsource.com/sonarqube instead of https://docs.sonarqube.org/
- Add deprecation notes about SonarScanner for Ant
- Update READMEs for minor typos/corrections
- Update link for Swift Coverage
- Update Sonar scanner versions for Maven, Gradle, .NET
- Update JaCoCo version
- Update SonarSource license year

### CODE REVIEW

File: doc/jacoco.md
Comment (joe-tingsanchali-sonarsource) [inline doc/jacoco.md:1]: Removed this folder and the Jacoco README. It was very much out-of-date. I didn't see a need to keep it here?

Maybe I can add a placeholder file here with a link to https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/test-coverage/java-test-coverage/?

**Review by fan-yang-sonarsource — APPROVED** [review]

File: sonar-scanner-ant/ant-coverage/build.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
 	<!-- Step 1: Import JaCoCo Ant tasks -->
 	<taskdef uri="antlib:org.jacoco.ant" resource="org/jacoco/ant/antlib.xml">
 		<!-- Update the following line with the location to the jacocoant.jar -->
-		<classpath path="path/to/jacoco-0.8.7/lib/jacocoant.jar" />
+		<classpath path="/Users/jting/Documents/jacocoant.jar" />
```
Comment (louis-chapuis-sonarsource) [inline sonar-scanner-ant/ant-coverage/build.xml]: Maybe it is better to leave a generic path here

File: sonar-scanner-ant/ant-coverage/build.xml [hunk 2]
```diff
@@ -94,7 +94,8 @@
 	<target name="sonar" depends="compile">
 		<taskdef uri="antlib:org.sonar.ant" resource="org/sonar/ant/antlib.xml">
 			<!-- Update the following line, or put the "sonar-ant-task-*.jar" file in your "$HOME/.ant/lib" folder -->
-			<classpath path="path/to/sonar/ant/task/lib/sonarqube-ant-task-*.jar" />
+			<!-- <classpath path="path/to/sonar/ant/task/lib/sonarqube-ant-task-*.jar" /> -->
+			<classpath path="/Users/jting/Documents/sonarqube-ant-task-2.7.1.1951.jar" />
```
Comment (louis-chapuis-sonarsource) [inline sonar-scanner-ant/ant-coverage/build.xml]: Maybe it is better to leave a generic path here

Comment (joe-tingsanchali-sonarsource) [reply sonar-scanner-ant/ant-coverage/build.xml]: Agree! Updated.

**Comment by kevin-hinz** [comment]
Just a quick comment that the name changes broke some links in the Docs that pointed to the sonar-scanning-examples.

**Pull request merged**